src/include/sof/audio/component.h

```diff
@@ -696,16 +696,25 @@ static inline struct comp_dev *comp_alloc(const struct comp_driver *drv,
 
 #if defined UNIT_TEST || defined __ZEPHYR__
 #define DECLARE_MODULE(init)
+#define DECLARE_TWO_MODULES(initfirst, initsecond)
 #elif CONFIG_LIBRARY
 /* In case of shared libs components are initialised in dlopen */
 #define DECLARE_MODULE(init) __attribute__((constructor)) \
 	static void _module_init(void) { init(); }
+#define DECLARE_TWO_MODULES(initfirst, initsecond) __attribute__((constructor)) \
+	static void _module_init(void) { initfirst(); initsecond(); }
```

Comment:
ah, this is the reason for this `DECLARE_TWO_MODULES()`! I see, sorry, I overlooked it. What if you had
```
#define DECLARE_MODULE(init) __attribute__((constructor)) \
	static void _module_init_##init(void) { init(); }
```
would it work then? You can have two functions with `__attribute__((constructor))` in a single file, cannot you?

Reply:
Yep, this is exactly the solution that I copied from Liam's plugin branch work. I'll push the new version next and make it non-draft. The CI test that previously failed passes now.